File: Editorial/0067_Add_Binary/Go/2_Bit_Manipulation.go
// Bit Manipulation is used to perform operations using bit wise XOR, AND and shift left.
func addBinary(a string, b string) string {
    var x, y, zero, carry, answer big.Int
    _, _ = x.SetString(a, 2)
    _, _ = y.SetString(b, 2)
    zero.SetInt64(0)
    for y.Cmp(&zero) != 0 {
        answer.Xor(&x, &y)
        carry.And(&x, &y)
        carry.Lsh(&carry, 1)
        x.Set(&answer)
        y.Set(&carry)
    }
    return x.Text(2)
}

File: Editorial/0000-0100/0067_Add_Binary/TypeScript/2_Bit_Manipulation.ts
// Bit Manipulation is used to perform operations using bit wise XOR, AND and shift left.
function addBinary(a: string, b: string): string {
    let x = BigInt(`0b${a}`);
    let y = BigInt(`0b${b}`);
    while (y) {
        let answer = x ^ y;
        let carry = (x & y) << BigInt(1);
        x = answer;
        y = carry;
    }
    return x.toString(2);
}
